site_scons/buildutils.py

```diff
@@ -407,302 +597,289 @@ def compareCsvFiles(env, file1, file2):
-        default = 'yes'
-    elif default is False:
-        default = 'no'
 
-    # First line: "* option-name: [ choice1 | choice2 ]"
-    lines = ['* %s: [ %s ]' % (opt.key, values)]
+    message = [
+        textwrap.dedent(
+            """
+                **************************************************
+                *   Configuration options for building Cantera   *
+                **************************************************
+
+        The following options can be passed to SCons to customize the Cantera
+        build process. They should be given in the form:
+
+            scons build option1=value1 option2=value2
+
+        Variables set in this way will be stored in the 'cantera.conf' file and reused
+        automatically on subsequent invocations of scons. Alternatively, the
+        configuration options can be entered directly into 'cantera.conf' before
+        running 'scons build'. The format of this file is:
+
+            option1 = 'value1'
+            option2 = 'value2'
+
+                **************************************************"""
+        )
+    ]
+
+    option_wrapper = textwrap.TextWrapper(
+        initial_indent=4 * " ",
+        subsequent_indent=4 * " ",
+        width=72,
+    )
+    bool_options = {True: "yes", False: "no"}
+    for opt in options.options:
+        # Extract the help description from the permitted values. Original format
+        # is in the format: "Help text (value1|value2)" for EnumVariable and
+        # BoolVariable types or "Help text" for other Variables
+        if opt.help.endswith(")"):
+            help, values = opt.help.rsplit("(", maxsplit=1)
+            values = values.rstrip(")").strip().replace("|", " | ")
+            if not values:
+                values = "string"
+        else:
+            help = opt.help
+            values = "string"
 
-    # Help text, wrapped and idented 4 spaces
-    lines.extend(optionWrapper.wrap(re.sub(r'\s+', ' ',help)))
+        # First line: "* option-name: [ choice1 | choice2 ]"
+        lines = [f"* {opt.key}: [ {values} ]"]
 
-    # Default value
-    lines.append('    - default: %r' % default)
+        # Help text, wrapped and indented 4 spaces
+        lines.extend(option_wrapper.wrap(re.sub(r"\s+", " ", help)))
 
-    # Get the actual value in the current environment
-    if opt.key in env:
-        actual = env.subst('${%s}' % opt.key)
-    else:
-        actual = None
+        # Default value
+        default = bool_options.get(opt.default, opt.default)
+        lines.append(f"    - default: {default!r}")
 
-    # Fix the representation of boolean options
-    if actual == 'True':
-        actual = 'yes'
-    elif actual == 'False':
-        actual = 'no'
+        # Get the actual value in the current environment
+        if opt.key in env:
+            actual = env.subst(f"${opt.key!s}")
+            if actual in ("True", "False"):
+                actual = bool_options.get(actual == "True", actual)
```

Comment:
I'm not convinced this is more readable than what was here.

Reply:
Agreed!